readme.txt
=== Feedblitz Email Subscription ===
Contributors: wptaskforce
Donate link: http://www.wptaskforce.com/feedblitz-email-subscription/
Tags: Feedblitz, email, feed, Mailer, newsletter, rss, shortcode, subscriber, subscription, widget
Requires at least: 3.5
Tested up to: 3.6
Stable tag: 0.3.2
License: GPLv2 or later
License URI: http://www.gnu.org/licenses/gpl-2.0.html

The best way to keep up with your content feed by placing a feedblitz email subscription widget on your site.

== Description ==

This WordPress plugin will give the best way to keep up with your content feed by placing a feedblitz email subscription on your wordpress site.
Feedblitz Email Subscription gives the best way to keep up with your content feed by placing a feedblitz email subscription on your wordpress site.
Easy to customize, just put your feed id and publisher id and your ready to lunch.
Shortcode usage:<strong>[feedblitz id="your_feed_id" publisher_id="your_publisher_id"]</strong>.

<h4>Key Features & Options</h4>
<ul>
<li>With additional admin panel</li>
<li>Template function</li>
<li>Support shortcode</li>
<li>Extensive widget options</li>
<li>Widget intro text and outro-text for your additional information, support HTML.</li>
</ul>

<h4>For Further Readings</h4>
If you have any trouble for setting the Feedblitz Email Subscription, please read this articles for getting it works perfectly as you wish.
<ul>
<li><a href="http://www.wptaskforce.com/feedblitz-email-subscription/">FeedBlitz Email Subscription</a></li>
<li><a href="http://www.wptaskforce.com/how-to-use-feedblitz-email-subscription/">How to Use FeedBlitz Email Subscription</a></li>
</ul>

== Installation ==

<p>Here's how to use Feedblitz Email Subscription:</p>
<ol>
<li><a href = "http://wordpress.org/plugins/feedblitz-email-subscription/" >Download </a>the plugin.</li>
<li>Upload the plugin zipped file to your desired wordpress site.</li>
<li>Activate the plugin.
<li>To use the widget, go to widget area look for - 'Feedblitz Email Subscription'.</li>
<li>To use the shortcode, go to the page where you want to add the shortcode , insert the code - <strong>[feedblitz id="your_feed_id" publisher_id="your_publisher_id"]</strong>.
<br/><br/>
***NOTE*** : <br/>
	These two values the 'id' and 'publisher_id' are very neccessary. <br/>
	The 'id' refers to your 'your_feed_id', and the other id refers to your "your_publisher_id".<br/><br/>

YOU CAN GET THESE TWO ID's BY FOLLOWING THESE STEPS:

<ol>
  <li>LOGIN to your Feedblitz account. Refer to this link => https://www.feedblitz.com</li>
  <li>Select and click the site where you want to put the feed.</li>
  <li>Look for the 'FORMCODE' button in the displayed thumbnails and CLICK IT.</li>
  <li>After that an html form will be showed,,,look for these two input field:
     <ul>
	<li>&lt;input name=&quot;FEEDID&quot; type=&quot;hidden&quot; value=&quot;11111&quot;&gt;</li>
	<li>&lt;input name=&quot;PUBLISHER&quot; type=&quot;hidden&quot; value=&quot;2222222&quot;&gt;</li>
    </ul>
  </li>
</ol>	
</li>
<li>Copy and paste each value respectively to your shortcode.</li>
<li>Finished.</li>
</ol>

== Screenshots ==

1. Widget Settings
2. Widget in frontend
3. Example of feedblitz.com settings


== Changelog ==

= 0.3.2 =
* Add Admin Settings
= 0.3.1 =
* Add Shortcode 
= 0.3.0 =
* Add widget option


== Upgrade Notice ==

Just upgrade via Wordpress.


Changelog.txt

7-20-2013      	Editable Instructions

5-30-2013      	Supports Feedblitz Email Subscription
		Popup subscription instead loading in tab.

lib/settings.php
<?php
add_action('admin_menu', 'feedblits_plugin_settings');
function feedblits_plugin_settings() {    
    add_menu_page('Feedlitz Settings', 'Feedlitz Settings', 'administrator', 'feedblitz_settings', 'feedblitz_display_settings');
}
function feedblitz_display_settings() {
	
 $feed_id = (get_option('FEEDID') != '') ? get_option('FEEDID') : '';
 $publisher_id = (get_option('PUBLISHER') != '') ? get_option('PUBLISHER') : '';
 
 
    $html = '<form method="post" name="feedblitz-options" action="options.php">
			<h3>Select Your Settings For Feedblitz</h3>' . wp_nonce_field('update-options') . '
			<label>Feed ID :</label><input type="text" placeholder="Enter Feed ID" name="FEEDID" value="' .  $feed_id . '" />            <label>Publisher ID : </label><input type="text" placeholder="Enter Publisher ID"name="PUBLISHER" 
			value="' . $publisher_id . '" /> 
				<input type="hidden" name="action" value="update" />  
				<input type="hidden" name="page_options" value="FEEDID,PUBLISHER" /> 
				<input type="submit" name="Submit" value="Update" />
			</form>';         
   echo $html;
}
?>

lib/dw_widget.php
<?php
class DW_Widget extends WP_Widget {  //openning class DW_Widget
    /**
     * Prints a basic text input for widget forms
     */
	  //insert code here
	   
		
		
		 
    function text_field($name, $value, $label = null) {
		
        if(!$label) { $label =  ucwords($name); }
		$label = explode('_', $label ); 
		?>
        
        <p>
            <label for="<?php echo $this->get_field_id($name); ?>">
                <?php foreach($label as $label){			
						echo $label.' ';
			}  ?>:
                <input class="widefat"
                       id="<?php echo $this->get_field_id($name); ?>"
                       name="<?php echo $this->get_field_name($name); ?>"
                       type="text" value="<?php echo $value; ?>" />
            </label>
        </p>
    <?php }
	 function text_area($name, $value, $label = null) {
        if(!$label) { $label =  ucwords($name); }
		$label = explode('_', $label ); ?>
        
        <p>
            <label for="<?php echo $this->get_field_id($name); ?>">
                <?php foreach($label as $label){			
						echo $label.' ';
					}  
					?>:
                <textarea class="widefat"
                       id="<?php echo $this->get_field_id($name); ?>"
                       name="<?php echo $this->get_field_name($name); ?>"
                       type="text"><?php echo $value; ?></textarea>
            </label>
        </p>
    <?php }
	
	
} //closing of class DW_Widget
?>


lib/feedblitz_email_subscription_shortcode.php
<?php
if (!defined("FEEDBLITZ_MAILVERIFY_URL")) {
  define("FEEDBLITZ_MAILVERIFY_URL", "http://www.feedblitz.com/f/f.fbz?AddNewUserDirect&ajax=4");
  
}


function feedblitz_shortcode( $atts ) {
	extract(shortcode_atts(array(
		'id'		=> '',
		'publisher_id'		=> '',		
		'instructions'	=> '',
		'text'	=>'',
		'submit'		=> 'Subscribe'
	), $atts ));
		
	$html='
    <form name="feedblitzform"  method="POST" target="popupwindow" action="'.FEEDBLITZ_MAILVERIFY_URL.'"  onsubmit="window.open(\''.FEEDBLITZ_MAILVERIFY_URL.'\', \'popupwindow\', \'scrollbars=yes,width=550,height=520\');return true" _lpchecked="1">
 <p class="sub_instruct"> '.$instructions .'</p>
 <p class="sub_instruct"> <input class="feedblitz-shortcode"  name="EMAIL" maxlength="64" type="text"  value="'.$text.'">
 <input name="EMAIL_" maxlength="64" type="hidden"  value="">
 <input name="EMAIL_ADDRESS" maxlength="64" type="hidden" size="25" value=""></p>
 <input name="FEEDID" type="hidden" value="'.$id.'">
 <input name="PUBLISHER" type="hidden" value="'.$publisher_id.'">
 <input type="submit" value="'.$submit.'">
</form>
  ';
	 return $html;
?>
<script language="Javascript">function feedblitzformi(){var x=document.getElementsByName('feedblitzform');for(i=0;i<x.length;i++){x[i].EMAIL.style.display='block';}}feedblitzformi();</script>
 
     
<?php 
}

add_shortcode('feedblitz', 'feedblitz_shortcode');

?>

lib/feedblitz_email_widget.php
<?php
add_action('widgets_init', create_function('', 'return register_widget("NotificationWidgetFeedblitz");'));

class NotificationWidgetFeedblitz extends DW_Widget {

// Setup class vars
	var $prefix;
	var $textdomain;

  function NotificationWidgetFeedblitz() {  // construct the widget Feedblitz() 
  
	  // Set up the widget options
        $this->prefix = 'feedblitz-email-subscription';
		$this->textdomain = 'feedblitz-email-subscription';	
		$widget_options = array(
			'classname' => 'feedblitz-email-subscription',
			'description' => esc_html__( 'The best way to keep up with your content feed by placing a feedblitz email subscription 			widget on your site.', $this->textdomain )
		);

		// Set up the widget control options
		$control_options = array(
			'width' => 'auto',
			'height' => 'auto',
			'id_base' => $this->prefix
		);
	
parent::WP_Widget(
	$this->prefix, esc_attr__( 'Feedblitz Email Subscription', $this->textdomain ), $widget_options,$control_options );
	
	//attached some css files
	if ( is_active_widget( false, false, $this->id_base ) && ! is_admin() ) {
			wp_enqueue_style("feedblitz-email-subscription", FEEDBLITZ_EMAIL_SUBSCRIPTION_URL . "css/feedblitz-widget.css");
		}
}//close constructor

  function widget($args, $instance) {
    extract( $args );
	$title            = apply_filters('widget_title', $instance['title']);	
	$text             = $instance['text'];
	$feed_id          = $instance['feed_id']?$instance['feed_id']:get_option('FEEDID') ;
	$publisher_id     = $instance['publisher_id']?$instance['publisher_id']:get_option('PUBLISHER');
	$instructions     = $instance['instructions'];
	$intro_text       = $instance['intro_text'];
	$submit           = $instance['submit']?$instance['submit']:'Subscribe';
	$outro_text       = $instance['outro_text'];	
	$json['site_url'] = get_bloginfo('url');
    echo $before_widget;
	
	/*	checks if feed id and publisher id are configured ,display the form if configured properly, else returns error message*/
	if ((!empty($feed_id) && !empty($publisher_id))){
	if ($title) echo $before_title . $title . $after_title;	 
?>
  
  
  <script>
  	var default_setting = <?php echo json_encode($json); ?>;
  	function openWindowSubscribe()
  	{  		
  		window.subscribeWindow = window.open('<?php echo FEEDBLITZ_MAILVERIFY_URL; ?>&ajax=4#0', 'popupwindow', 'scrollbars=yes,width=550,height=520');
  	}

  	function checkWin()
  	{
  		if (window.subscribeWindow)
  		{
  			if (!window.subscribeWindow || window.subscribeWindow.closed)
  			{
  				window.subscribeWindow = false;  				
  				window.location = default_setting.site_url + '/thankyou';
  			}
  		}
  	}
  	window.setInterval("checkWin()",3*1000);  	
  </script>

  <form name="feedblitzform"  method="POST" target='popupwindow' action="<?php echo FEEDBLITZ_MAILVERIFY_URL; ?>&ajax=4#0"  onsubmit="openWindowSubscribe(); return true" _lpchecked="1">
 <p class="sub_instruct"><?php echo $instructions; ?></p>
 <p class="sub_instruct"><?php echo $intro_text; ?></p>
 <p class="feedblitz-email-subscription"> <input style="display:none" name="EMAIL" maxlength="64" type="text"  value="<?php echo $text; ?>" onblur="if(this.value == '') { this.value='<?php echo $text; ?>'}" onfocus="if (this.value == '<?php echo $text; ?>') {this.value=''}">
 <input name="EMAIL_" maxlength="64" type="hidden"  value="">
 <input name="EMAIL_ADDRESS" maxlength="64" type="hidden" size="25" value=""></p>
 <input name="FEEDID" type="hidden" value="<?php  echo $feed_id; ?>">
 <input name="PUBLISHER" type="hidden" value="<?php echo $publisher_id; ?>">
 <input type="submit" value="<?php echo $submit; ?>">
  <p class="sub_instruct"><?php echo $outro_text; ?></p>
</form>



<script language="Javascript">function feedblitzformi(){var x=document.getElementsByName('feedblitzform');for(i=0;i<x.length;i++){x[i].EMAIL.style.display='block';}}feedblitzformi();</script>

  
  
  <?php 
  
  }else{
	  echo '<span style="background-color:#FF0000;color:#fff;padding:20px 15px;font-size:12px">WARNING!,Feed ID and Publisher ID Required!</span>';	  
	  }
	  
  echo $after_widget;
  
  }

  function update($new_instance, $old_instance) {
    $instance = $old_instance;
    $instance['title'] = strip_tags($new_instance['title']);
	$instance['text'] = $new_instance['text'];
	$instance['instructions'] = $new_instance['instructions'];
    $instance['feed_id'] = $new_instance['feed_id'];
	$instance['publisher_id'] = $new_instance['publisher_id'];
	$instance['intro_text'] = $new_instance['intro_text'];
	$instance['outro_text'] = $new_instance['outro_text'];
	$instance['submit'] = $new_instance['submit']; 
	
    return $instance;
  }

  function form($instance) {
	 // Set up the default form values
		$defaults = array(
			'title' 		=> __( 'Email Subscription', $this->textdomain ),
			'text' 			=> __( 'Your email here', $this->textdomain ),
			'submit' 		=> __( 'Subscribe', $this->textdomain ),
			'intro_text'	=> '',
			'outro_text' 	=> ''
		);

		// Merge the user-selected arguments with the defaults
		$instance = wp_parse_args( (array) $instance,$defaults); 
         extract($instance);
		 
	
	
    $this->text_field("title", $title);
	$this->text_field("text", $text); 
	$this->text_area("instructions", $instructions);
    $this->text_field("feed_id", $feed_id);
	$this->text_field("publisher_id", $publisher_id);
	$this->text_area("intro_text", $intro_text); 
	$this->text_area("outro_text", $outro_text); 
	$this->text_field("submit", $submit); 
	
  }
  
  
}
?>